service/s3/internal/customizations/update_endpoint_test.go

```diff
@@ -442,31 +442,28 @@ func TestEndpointWithARN(t *testing.T) {
 			expectedSigningName:   "s3-outposts",
 			expectedSigningRegion: "us-gov-east-1",
 		},
-		"Outpost AccessPoint Fips region": {
+		"Outpost AccessPoint FIPS cross-region": {
 			bucket: "arn:aws-us-gov:s3-outposts:us-gov-east-1:123456789012:outpost:op-01234567890123456:accesspoint:myaccesspoint",
 			options: s3.Options{
 				Region: "fips-us-gov-west-1",
 			},
 			expectedErr: "ConfigurationError : client region does not match provided ARN region",
 		},
-		"Outpost AccessPoint Fips region in Arn": {
-			bucket: "arn:aws-us-gov:s3-outposts:fips-us-gov-west-1:123456789012:outpost:op-01234567890123456:accesspoint:myaccesspoint",
+		"Outpost AccessPoint with FIPS client cross-region": {
+			bucket: "arn:aws-us-gov:s3-outposts:us-gov-east-1:123456789012:outpost:op-01234567890123456:accesspoint:myaccesspoint",
 			options: s3.Options{
-				Region:          "fips-us-gov-west-1",
-				EndpointOptions: endpoints.Options{DisableHTTPS: true},
-				UseARNRegion:    true,
+				Region:       "fips-us-gov-west-1",
+				UseARNRegion: true,
 			},
-			expectedErr: "InvalidARNError : resource ARN not supported for FIPS region",
+			expectedErr: "use of ARN is not supported when client or request is configured for FIPS",
```

Comment:
These two tests seem similar enough but the error returned is different.

Reply:
This is because the check to see if `UseARNRegion` is enabled/disabled is before the FIPS check for outpost ARNs